Makefile

```diff
@@ -1,149 +1,152 @@
+clean:
+	if [ -d "$(OUTDIR)" ]; then rmdir $(OUTDIR); fi
+	if [ -d "$(CACHEDIR)" ]; then rmdir $(CACHEDIR); fi
+	if [ -d "$(REPO)" ]; then rm -r $(REPO); fi
+.PHONY: clean
 
-install-dependencies: add-repo
-	for dep in $(FDO_DEPS) $(GNOME_DEPS); do \
-		flatpak install -y --user flathub $$dep ; \
-		flatpak update -y --user $$dep ; \
-	done
-	flatpak list --user --runtime --show-details
+export: ;
+.PHONY: export
 
-install-locale-dependencies: add-repo
-	for dep in $(LOCALE_DEPS); do \
-		flatpak uninstall -y --user $$dep ; \
-		flatpak install -y --user flathub $$dep ; \
-	done
-	flatpak list --user --runtime --all --show-details
+sign: export
+.PHONY: sign
 
-clean-dependencies: add-repo
-	flatpak uninstall -y --user $(FDO_DEPS)
-	flatpak uninstall -y --user $(GNOME_DEPS)
+bundle: ;
+.PHONY: bundle
 
-check: com.endlessm.apps.Sdk.json
-	@echo "  CHK   $<"; json-glib-validate com.endlessm.apps.Sdk.json
+fetch-junctions:
+	$(BST) $(BST_ARGS) fetch freedesktop-sdk.bst gnome-sdk.bst
+.PHONY: fetch-junctions
 
-clean:
-	@rm -rf builddir 
-	@rm -f ${SUBST_FILES}
-	@rm -f com.endlessm.apps.Sdk.json
-	@rm -f com.endlessm.apps.Sdk.json.in
-
-maintainer-clean: clean
-	@rm -rf .flatpak-builder
-	@rm -rf ${REPO}
-
-import-artefacts:
-	@if [ -f ${REPO}.tar ]; then \
-	    tar xf ${REPO}.tar && \
-	    ostree fsck --repo=${REPO} ; \
-	    rm -f ${REPO}.tar
-	fi
-
-bundle-artefacts:
-	@tar cf ${REPO}.tar ${REPO}
-
-update-fonts:
-	@for font in $(GOOGLE_FONTS); do \
-		wget https://fonts.google.com/download?family=$$font -O fonts/$$font.zip; \
-	done
-
-.PHONY: sdk icontheme add-repo install-dependencies clean-dependencies maintainer-clean bundle-artefacts
+update-refs:
+	$(BST) $(BST_ARGS) track freedesktop-sdk.bst gnome-sdk.bst
+	$(BST) $(BST_ARGS) track flatpak-runtimes.bst flatpak-platform-extensions.bst --deps=all
+.PHONY: update-refs
+
+
+$(OUTDIR):
+	mkdir -p $(OUTDIR)
+
+$(CACHEDIR):
+	mkdir -p $(CACHEDIR)
+
+$(REPO):
+	$(OSTREE) init --repo=$(REPO) --mode=bare-user-only
+
+
+.git/hooks/pre-commit: utils/git-pre-commit
+	ln -frs $< .git/hooks/pre-commit
+
+
+flatpak-version.yml: $(GIT_HOOKS) | CLEAN-flatpak-version.yml
+	./utils/generate-version $@
+.PHONY: flatpak-version.yml
+
+CLEAN-flatpak-version.yml:
+	$(GIT) checkout HEAD flatpak-version.yml
+.PHONY: CLEAN-flatpak-version.yml
+clean: CLEAN-flatpak-version.yml
+
+
+BUILD-flatpak-runtimes: flatpak-version.yml elements/**/*.bst
+	$(BST) $(BST_ARGS) build flatpak-runtimes.bst
+.PHONY: BUILD-flatpak-runtimes
+
+CHECK-flatpak-runtimes: flatpak-version.yml | fetch-junctions
+	$(BST) $(BST_ARGS) show flatpak-runtimes.bst
+.PHONY: CHECK-flatpak-runtimes
+check: CHECK-flatpak-runtimes
+
+$(FLATPAK_RUNTIMES_REPO): BUILD-flatpak-runtimes | $(CACHEDIR)
+	$(BST) $(BST_ARGS) checkout --hardlinks --force flatpak-runtimes.bst $@
+
+EXPORT-$(FLATPAK_RUNTIMES_REPO): $(FLATPAK_RUNTIMES_REPO) | $(REPO)
+	$(OSTREE) pull-local --repo=$(REPO) $(FLATPAK_RUNTIMES_REPO)
+	$(OSTREE) summary --repo=$(REPO) --update
```

Comment:
You'll also need to sign the ostree summary in order to publish it. Maybe you want to just keep this and the commit signing in the generic `export` target, keep the `EXPORT_ARGS` as before and make the commit signing conditional on it being set. Something like this:
```
EXPORT_ARGS ?=
export:
ifneq ($(EXPORT_ARGS),)
	flatpak build-sign $(EXPORT_ARGS) $(REPO)
endif
	flatpak build-update-repo $(EXPORT_ARGS) $(REPO)
```
`EXPORT_ARGS` really means `GPG_SIGN_ARGS`, but this is just using the same name that the jenkins script uses. One advantage of just using a bare variable is that it allows passing in a `--gpg-homedir` if necessary. Currently we don't do that, but with your changes it wouldn't be possible.

Then all the `EXPORT-*` targets would just do `pull-local` as prereqs for the generic `export` target.

Or, another way to manage the keys and homedir:
```
GPG_KEY_IDS ?=
GPG_HOMEDIR ?=
GPG_ARGS = $(strip $(patsubst %,--gpg-sign=%,$(GPG_KEY_IDS)) \
	$(patsubst %,--gpg-homedir=%,$(GPG_HOMEDIR)))
export:
ifneq ($(GPG_ARGS),)
	flatpak build-sign $(GPG_ARGS) $(REPO)
endif
	flatpak build-update-repo $(GPG_ARGS) $(REPO)
```

Reply:
Okay, fixed that now. Thank you for the detailed explanation!